specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/restorePoints.json

```diff
@@ -44,15 +44,15 @@
           "200": {
             "description": "Successfully returned restore points.",
             "schema": {
-              "$ref": "#/definitions/DatabaseRestorePointListResult"
+              "$ref": "#/definitions/RestorePointListResult"
             }
           },
           "default": {
             "description": "*** Error Responses: ***\n\n * 400 FeatureDisabledOnSelectedEdition - User attempted to use a feature which is disabled on current database edition.\n\n * 404 DatabaseRestorePointNotFound - Can not find database restore point.\n\n * 404 SubscriptionDoesNotHaveServer - The requested server was not found\n\n * 404 DatabaseDoesNotExist - User has specified a database name that does not exist on this server instance."
           }
         },
         "x-ms-pageable": {
-          "nextLinkName": "nextLink"
+          "nextLinkName": null
```

Comment:
why? This is not what `RestorePointListResult` looks like to me!

Reply:
Without this, it is causing breaking change in sdk. It introduces new type RestorePointListResult instead of IEnumerable<RestorePoint> as return type of List API.

Reply:
@jaredmoo

Reply:
@sarvesh-s lol ok, but note that this `IEnumerable` will only ever contain the first page of the operation! I'm fully aware how much this sucks and (being the C# generator guy) this is super embarassing, we could easily auto-fetch pages from an `IEnumerable` (we have the code!) but we literally have SDK consumers depending on the IEnumerable only being the first page. Hmmm okay, so if you're okay with it being only the first page, fine, otherwise we'll have to customize the SDK generation project (however, it's probably not just C#, so we're talking about a good bunch of customization). 😞

Reply:
I believe first page seem to be working OK so far, I mean entire list must be fitting in old API - if you mean memory size here.

Reply:
well, I mean whatever your service returns as a single page - if `RestorePoints_ListByDatabase` ever returns something with `nextLink` pointing to another page, that other page will not be accessible through the SDK. So, say, if there are 5000 restore points for a database, will your API return all of those in the `value` of the result? Or will it be split across pages?

Reply:
We can not have more than 84 restore points. So I think we are good.